packages/openapi-v3/test/integration/controller-spec.test.ts

```diff
@@ -70,49 +70,56 @@ describe('controller spec', () => {
 
     expect(getControllerSpec(FooController)).to.eql(expectedSpec);
   });
+
   it('does not produce nested definitions', () => {
     const paramSpec: ParameterObject = {
       name: 'foo',
       in: 'query',
     };
+
     @model()
     class Foo {
       @property() bar: number;
     }
+
     @model()
     class MyParam {
       @property() name: string;
       @property() foo: Foo;
     }
+
     class MyController {
       @post('/foo')
       foo(@param(paramSpec) body: MyParam) {}
     }
 
-    const schemas = (getControllerSpec(MyController).components || {})
-      .schemas as SchemasObject;
+    const components = getControllerSpec(MyController).components || {};
+    const schemas = components.schemas;
+
     expect(schemas).to.have.keys('MyParam', 'Foo');
-    expect(schemas.MyParam).to.not.have.key('definitions');
+    expect((schemas || {}).MyParam).to.not.have.key('definitions');
```

Comment:
Please move `|| {}` to the line where `const schemas` is built.

However, I think this is redundant. When the previous assertion `expect(schema).to.have.keys()` passes, then surely `schemas` is a non-empty object value?

Reply:
@bajtos I add `(schemas || {})` since editor complains: 
>`schemas` could be undefined

The previous `expect` infers `schemas` is not `undefined` but the editor doesn't know....

Reply:
Ah, I think you can use `schemas!.MyParam` in such case.